internal/resources/fleetpolicy.go

```diff
@@ -0,0 +1,458 @@
+	}
+
+	policyTemplateName := packagePolicy.TemplateName
+	if policyTemplateName == "" {
+		name, err := findPolicyTemplateForInputPackage(manifest, packagePolicy.InputName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to determine the associated policy_template: %w", err)
+		}
+		policyTemplateName = name
+	}
+
+	policyTemplate, err := selectPolicyTemplateByName(manifest.PolicyTemplates, policyTemplateName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find the selected policy_template: %w", err)
+	}
+
+	ds := kibana.PackageDataStream{
+		Name:      packagePolicy.Name,
+		Namespace: policy.Namespace,
+		PolicyID:  policy.ID,
+		Enabled:   !packagePolicy.Disabled,
+		Inputs: []kibana.Input{
+			{
+				PolicyTemplate: policyTemplate.Name,
+				Enabled:        !packagePolicy.Disabled,
+			},
+		},
+	}
+	ds.Package.Name = manifest.Name
+	ds.Package.Title = manifest.Title
+	ds.Package.Version = manifest.Version
+
+	streamInput := policyTemplate.Input
+	ds.Inputs[0].Type = streamInput
+
+	dataset := fmt.Sprintf("%s.%s", manifest.Name, policyTemplate.Name)
+	streams := []kibana.Stream{
+		{
+			ID:      fmt.Sprintf("%s-%s.%s", streamInput, manifest.Name, policyTemplate.Name),
+			Enabled: !packagePolicy.Disabled,
+			DataStream: kibana.DataStream{
+				Type:    policyTemplate.Type,
+				Dataset: dataset,
+			},
+		},
+	}
+
+	// Add policyTemplate-level vars.
+	vars := setKibanaVariables(policyTemplate.Vars, common.MapStr(packagePolicy.Vars))
+	if _, found := vars["data_stream.dataset"]; !found {
+		var value packages.VarValue
+		value.Unpack(dataset)
+		vars["data_stream.dataset"] = kibana.Var{
+			Value: value,
+			Type:  "text",
+		}
+	}
+
+	streams[0].Vars = vars
+	ds.Inputs[0].Streams = streams
+
+	return &ds, nil
+}
+
+func setKibanaVariables(definitions []packages.Variable, values common.MapStr) kibana.Vars {
+	vars := kibana.Vars{}
+	for _, definition := range definitions {
+		val := definition.Default
+
+		value, err := values.GetValue(definition.Name)
+		if err == nil {
+			val = packages.VarValue{}
+			val.Unpack(value)
+		}
+
+		vars[definition.Name] = kibana.Var{
+			Type:  definition.Type,
+			Value: val,
+		}
+	}
+	return vars
+}
+
+func (f *FleetAgentPolicy) Update(ctx resource.Context) error {
+	if f.Absent {
+		provider, err := f.provider(ctx)
+		if err != nil {
+			return err
+		}
+
+		err = provider.Client.DeletePolicy(ctx, f.ID)
+		if err != nil {
+			return fmt.Errorf("could not delete policy %q: %w", f.Name, err)
+		}
+
+		return nil
+	}
+
+	return errors.New("update not implemented")
+}
+
+type FleetAgentPolicyState struct {
+	current  *kibana.Policy
+	expected bool
+}
+
+func (s *FleetAgentPolicyState) Found() bool {
+	return !s.expected || s.current != nil
+}
+
+func (s *FleetAgentPolicyState) NeedsUpdate(resource resource.Resource) (bool, error) {
+	policy := resource.(*FleetAgentPolicy)
+	return policy.Absent == (s.current == nil), nil
```

Comment:
Here, if the policy is set as absent and the package does not exist (s.current == nil), it will return that it needs an update. Should it return `false` for that case?

Maybe I'm missing something, should it return just `true` for these cases ? 
- `policy.Absent == false` and `s.current == nil`
- `policy.Absent == true` and `s.current != nil`

If so, I was thinking that this condition should be like:
```suggestion
	return policy.Absent == (s.current != nil), nil
```

Reply:
Yes, I think you are right, I will revisit this.

Reply:
Fixed in https://github.com/elastic/elastic-package/pull/1836/commits/668231a651cba0c35c486bfe7914dc65729b575d.